platform/mbed_retarget.cpp

```diff
@@ -120,6 +121,17 @@ static void init_serial() {
 #endif
 }
 
+/**
+ * Sets errno when file opening fails.
+ * Wipes out the filehandle too.
+ */
+static int handle_open_errors(int error, unsigned filehandle_idx) {
+    errno = -error;
```

Comment:
@geky will be able to answer this better. For some reason error codes returned by C library are negative. So when you return an error, you add a minus sign to errno value to indicate error plus indication of errno.